### PR TITLE
Default implementation of handleInvalidEmptyPreconditionCase added in AbtractFormattedChangeLogParser to avoid breaking extensions

### PR DESCRIPTION
**Pull request by MalloD12**
## Impact

- [x] Bug fix (non-breaking change which fixes expected existing functionality)
- [ ] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 

## Description

Recently with the merge of this [PR](https://github.com/liquibase/liquibase/pull/5456/) the team has noticed this is breaking extensions, so just move the code implementation of `handleInvalidEmptyPreconditionCase` to the abstract class which should fix this issue if no extension overrides this method.

### CODE REVIEW

**Comment by vitaliimak** [comment]
Thanks, @MalloD12 for such a quick fix!
It should work, but, I'm wondering if it is possible to leave the abstract class without SQL-specific implementation. The commercial MongoDB extension does not use `sql-check`, `table-exists`, and so on.
Is it possible to write something like this?
```
protected void handleInvalidEmptyPreconditionCase(ChangeLogParameters changeLogParameters, ChangeSet changeSet, Matcher preconditionMatcher) throws ChangeLogParseException {
    throw new NotImplementedException();
}
```
And left SQL-specific code in the `FormattedSqlChangeLogParser` class.

**Review by StevenMassaro — APPROVED** [review]

**Comment by MalloD12** [comment]
> protected void handleInvalidEmptyPreconditionCase(ChangeLogParameters changeLogParameters, ChangeSet changeSet, Matcher preconditionMatcher) throws ChangeLogParseException {
>     throw new NotImplementedException();
> }

Ohh, I see. I get it now. Sure, I can do that @vitaliimak.

**Review by vitaliimak — APPROVED** [review]

File: liquibase-standard/src/main/java/liquibase/parser/AbstractFormattedChangeLogParser.java [hunk 1]
```diff
@@ -610,6 +609,10 @@ protected InputStream openChangeLogFile(String physicalChangeLogLocation, Resour
         return resourceAccessor.getExisting(physicalChangeLogLocation).openInputStream();
     }
 
+    protected void handleInvalidEmptyPreconditionCase(ChangeLogParameters changeLogParameters, ChangeSet changeSet, Matcher preconditionMatcher) throws ChangeLogParseException {
+        throw new NotImplementedException();
```
Comment (filipelautert) [inline liquibase-standard/src/main/java/liquibase/parser/AbstractFormattedChangeLogParser.java:613]: Please add a message explaining what we reached, like: 
```
throw new NotImplementedException("Invalid empty precondition found");
```

**Review by filipelautert — CHANGES_REQUESTED** [review]

File: liquibase-standard/src/main/java/liquibase/parser/AbstractFormattedChangeLogParser.java (hunk 1 above)
Comment (MalloD12) [reply liquibase-standard/src/main/java/liquibase/parser/AbstractFormattedChangeLogParser.java]: Done

**Review by filipelautert — APPROVED** [review]

**Pull request merged**